src/components/Map/EditableMap/EditableMap.js

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import MapGL from 'react-map-gl'
+import { Editor, DrawPolygonMode, EditingMode } from 'react-map-gl-draw'
+import Icon from '../../Icon'
+import { mapbox } from 'config'
+
+import { getEditHandleStyle, getFeatureStyle } from './EditableMapStyles'
+import './EditableMap.scss'
+
+export default function EditableMap (props) {
+  const { locationObject, polygon, savePolygon } = props
+  const centerAt = locationObject?.center
+  const showLocationOnMap = !polygon && centerAt
+  const [viewport, setViewport] = useState(null)
+  const [mode, setMode] = useState(null)
+  const [selectedFeatureIndex, setSelectedFeatureIndex] = useState(null)
+  const [isModeDrawing, setIsModeDrawing] = useState(false)
+  const editorRef = useRef(null)
+  const existingPolygon = polygon ? [{
+    geometry: polygon,
```

Comment:
this is the only variable that changes, so less code to have the ternary operator only here for this line